### PR TITLE
documentation update to demonstrate mutability

### PR DESCRIPTION
**Pull request by toidiu**
#44467 

- demonstrate correct implementation returns `Some`
- demonstrate out of bounds returns `None`
- demonstrate mutability

### CODE REVIEW

**Comment by toidiu** [comment]
r? frewsxcv

**Comment by toidiu** [comment]
#44467

File: src/liballoc/str.rs
Comment (frewsxcv) [inline src/liballoc/str.rs]: alternatively, something like this could maybe work here, but what you have is :cool: too

```rust
if let Option(s) = v.get_mut(0..2) {
    s.make_ascii_uppercase();
}
```

**Review by frewsxcv — APPROVED** [review]
lgtm! will wait for travis before approving for merge. thanks!

**Pull request merged**
